public/crclib.c

```diff
@@ -112,65 +112,42 @@ void CRC32_ProcessByte( dword *pulCRC, byte ch )
 void CRC32_ProcessBuffer( dword *pulCRC, const void *pBuffer, int nBuffer )
 {
 	dword	ulCrc = *pulCRC, tmp;
-	byte	*pb = (byte *)pBuffer;
-	uint	nFront;
-	int	nMain;
-JustAfew:
-	switch( nBuffer )
+	dword	*pb = (dword *)pBuffer;
+	byte	*pbc = NULL;
+
+	while(nBuffer >= sizeof(long long))
 	{
-	case 7: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 6: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 5: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 4:
-		memcpy( &tmp, pb, sizeof(dword));
-		ulCrc ^= tmp;	// warning, this only works on little-endian.
-		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
+		ulCrc ^= LittleLong(*pb++);   // warning, this only works on little-endian.
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
+		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);	
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
-		*pulCRC = ulCrc;
-		return;
-	case 3: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 2: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 1: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 0: *pulCRC = ulCrc;
-		return;
-	}
-
-	// We may need to do some alignment work up front, and at the end, so that
-	// the main loop is aligned and only has to worry about 8 byte at a time.
-	// The low-order two bits of pb and nBuffer in total control the
-	// upfront work.
-	nFront = ((uint)pb) & 3;
-	nBuffer -= nFront;
-
-	switch( nFront )
-	{
-	case 3: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 2: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	case 1: ulCrc  = crc32table[*pb++ ^ (byte)ulCrc] ^ (ulCrc >> 8); // fallthrough
-	}
-
-	nMain = nBuffer >> 3;
-	while( nMain-- )
-	{
-		memcpy( &tmp, pb, sizeof(dword));
-		ulCrc ^= tmp;	// warning, this only works on little-endian.
+		ulCrc ^= LittleLong(*pb++); // warning, this only works on little-endian.
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
 		ulCrc  = crc32table[(byte)ulCrc] ^ (ulCrc >> 8);
-		memcpy( &tmp, pb + 4, sizeof(dword));
-		ulCrc ^= tmp; // warning, this only works on little-endian.
+		nBuffer -= sizeof(long long);
```

Comment:
Use integer64, C89 does not have long long.
https://github.com/FWGS/xash3d-fwgs/blob/master/common/xash3d_types.h#L32